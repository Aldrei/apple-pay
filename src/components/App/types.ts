export interface IApp {
  message: string;
}
