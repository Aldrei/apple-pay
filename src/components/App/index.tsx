import './App.css';
import { IApp } from './types';

function App({ message }: IApp) {
  return (
    <div className="App">
      <header className="App-header">
        <p>{message}</p>
        <a
          className="App-link"
          href="https://applepaydemo.apple.com/payment-request-api"
          target="_blank"
          rel="noopener noreferrer"
        >
          Apple Pay Demo
        </a>
        <a
          className="App-link"
          href="https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForms/simulation"
          target="_blank"
          rel="noopener noreferrer"
        >
          VTEX API
        </a>
      </header>
    </div>
  );
}

export default App;
