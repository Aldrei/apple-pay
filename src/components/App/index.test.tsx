import { render, screen } from '@testing-library/react';
import App from '.';

test('renders learn react link', () => {
  render(<App message="test" />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
