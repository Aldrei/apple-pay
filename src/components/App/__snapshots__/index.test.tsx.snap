// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders text param for link 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <p>
          test
        </p>
        <a
          class="App-link"
          href="https://applepaydemo.apple.com/payment-request-api"
          rel="noopener noreferrer"
          target="_blank"
        >
          Apple Pay Demo
        </a>
        <a
          class="App-link"
          href="https://developers.vtex.com/docs/api-reference/checkout-api#post-/api/checkout/pub/orderForms/simulation"
          rel="noopener noreferrer"
          target="_blank"
        >
          VTEX API
        </a>
      </header>
    </div>
  </div>
</body>
`;
